src/engine/Join.cpp

```diff
@@ -145,6 +150,16 @@ void Join::computeResult(ResultTable* result) {
                   leftRes->_idTable, _leftJoinCol, rightRes->_idTable,
                   _rightJoinCol, &result->_idTable);
 
+  /* TODO This is needed for the pre-compilation of the template hashJoin.
+   * Because the compiler doesn't know, that rand() < 0 can never be true,
+   * it compiles the used templates without breaking the actual join operation.
+  */
+  if (rand() < 0) {
+    CALL_FIXED_SIZE((std::array{lwidth, rwidth, reswidth}), &Join::hashJoin, this,
+                    leftRes->_idTable, _leftJoinCol, rightRes->_idTable,
+                    _rightJoinCol, &result->_idTable);
+  }
+
```

Comment:
Did you mean, that I should use the new `hashJoin` for the tests, or `hashJoinImpl`?
Because if it's `hashJoin`, I would first need to re-write most of the infrastructure for the join tests.

Reply:
But what about the callFixedSize in `useJoinFunctionOnIdTables`? Wouldn't there be problems, because hashJoinLambda has no template parameters?

Reply:
`hashJoinLambda` still has template parameters, it is just not using them internally. That's why it can still be used with `callFixedSize` to have a general interface for the tests.

Reply:
In spite of me misunderstanding a few things and making some rather embarissing mental errors, I managed to implement your advice. Now the release build compiles again and everything seems to be running correctly. (See [3f328dd](https://github.com/ad-freiburg/qlever/pull/821/commits/3f328dd02a86c71232c45eb7a3c329b85848e7b2).)